### ISSUE TITLE
Calling processor.process() twice results in django.db.utils.InternalError: no such savepoint

### ISSUE THREAD

**Issue by werwty** [project-koku/koku#1291]
**Describe the bug**
When writing unittests for the `GCPReportProcessor` I noticed that calling `process()` twice with the same processor instance results in an error. 
This is not a problem if I declare another instance of the processor and call process() again. 
This is also a problem with the `AWSReportProcessor` and the `AzureReportProcessor`.

**To Reproduce**
1. apply diff to `masu.test.processor.azure.test_azure_report_processor.AzureReportProcessorTest.test_azure_process_duplicates`
```
diff --git a/koku/masu/test/processor/azure/test_azure_report_processor.py b/koku/masu/test/processor/azure/test_azure_report_processor.py
index e37e9edc..eb6665d3 100644
--- a/koku/masu/test/processor/azure/test_azure_report_processor.py
+++ b/koku/masu/test/processor/azure/test_azure_report_processor.py
@@ -206,12 +206,12 @@ class AzureReportProcessorTest(MasuTestCase):
         with schema_context(self.schema):
             self.accessor._get_db_obj_query(AZURE_REPORT_TABLE_MAP['line_item']).delete()
 
-        processor = AzureReportProcessor(
-            schema_name=self.schema,
-            report_path=self.test_report,
-            compression=UNCOMPRESSED,
-            provider_id=self.azure_provider.id,
-        )
+        # processor = AzureReportProcessor(
+        #     schema_name=self.schema,
+        #     report_path=self.test_report,
+        #     compression=UNCOMPRESSED,
+        #     provider_id=self.azure_provider.id,
+        # )
         # Process for the second time
         processor.process()
```
2. run `./koku/manage.py test masu.test.processor.azure.test_azure_report_processor.AzureReportProcessorTest.test_azure_process_duplicates`

**Expected behavior**
I would expect processing with the same processor twice to not throw an error. If all rows have already been inserted into the DB then it should override all lineitems/not insert the lineitems again.

**However, note that we don't normally operate this way; this appears to only affect unit tests. Prioritize accordingly!**

**Additional context**

```
Traceback (most recent call last):
  File "/home/bizhang/.local/share/virtualenvs/koku-TAuOYtlY/lib/python3.6/site-packages/django/db/backends/utils.py", line 82, in _execute
    return self.cursor.execute(sql)
psycopg2.errors.InvalidSavepointSpecification: no such savepoint


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/bizhang/devel/koku/koku/masu/test/processor/gcp/test_gcp_report_processor.py", line 168, in test_gcp_process_twice
    self.processor.process()
  File "/home/bizhang/devel/koku/koku/masu/processor/gcp/gcp_report_processor.py", line 204, in process
    self._save_to_db(temp_table, report_db)
  File "/home/bizhang/devel/koku/koku/masu/processor/report_processor_base.py", line 125, in _save_to_db
    columns)
  File "/home/bizhang/devel/koku/koku/masu/database/report_db_accessor_base.py", line 178, in bulk_insert_rows
    transaction.savepoint_commit(KokuDBAccess._savepoints.pop())
  File "/home/bizhang/.local/share/virtualenvs/koku-TAuOYtlY/lib/python3.6/site-packages/django/db/transaction.py", line 65, in savepoint_commit
    get_connection(using).savepoint_commit(sid)
  File "/home/bizhang/.local/share/virtualenvs/koku-TAuOYtlY/lib/python3.6/site-packages/django/db/backends/base/base.py", line 359, in savepoint_commit
    self._savepoint_commit(sid)
  File "/home/bizhang/.local/share/virtualenvs/koku-TAuOYtlY/lib/python3.6/site-packages/django/db/backends/base/base.py", line 308, in _savepoint_commit
    cursor.execute(self.ops.savepoint_commit_sql(sid))
  File "/home/bizhang/.local/share/virtualenvs/koku-TAuOYtlY/lib/python3.6/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/bizhang/.local/share/virtualenvs/koku-TAuOYtlY/lib/python3.6/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/bizhang/.local/share/virtualenvs/koku-TAuOYtlY/lib/python3.6/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/home/bizhang/.local/share/virtualenvs/koku-TAuOYtlY/lib/python3.6/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/bizhang/.local/share/virtualenvs/koku-TAuOYtlY/lib/python3.6/site-packages/django/db/backends/utils.py", line 82, in _execute
    return self.cursor.execute(sql)
django.db.utils.InternalError: no such savepoint
```

**Comment by maskarb** [comment]
The `no such savepoint` error was addressed in #1433. However, the processing workflow was not been addressed.

**Comment by lcouzens** [comment]
The change is not _really_ testable so closing this as is.
